components/Footer.js

```diff
@@ -0,0 +1,26 @@
+import AddButton from './AddButton';
+import AddForm from './AddForm';
+import { useState } from 'react';
+import styled from 'styled-components';
+
+export default function Footer({ onAddItem }) {
```

Comment:
I don't fully understand why you are using a Footer as the container for your Form. In my opinion you could delete the Footer and directly add the Addform to your index.js 
A Footer is usually used for the impressum, links to social media or similar, some devs like to add their navbar to the footer, but you don't necessarily need a footer for your app. 
So you could transfer all this code, including the state and the conditional rendering into your AddForm component. This way you could avoid this extra layer and passing your state up and down :)